### PR TITLE
Feature/ab#45 keycloak config

### PR DESCRIPTION
**Pull request by witchpou**
Covers tickes ab#44 and ab#45. Authorization and Authentication via keycloak is implemented. 

## Description

It contains:
* realm with the following roles can be imported into keycloak: ROLE_admin, ROLE_user, ROLE_reader
* swagger is working
* logout button
* fix for csrf in backend
* user und role management is possible via keycloak
* github can be included as identity provider in keycloak
* getting group via api/usermanagement/userinfo

## Motivation and Context

Nasty authentication with Spring Boot.

## How has this been tested?

Manual tests with users admin, user, reader were done.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: application/pom.xml [hunk 1]
```diff
@@ -20,6 +20,10 @@
             <groupId>de.starwit</groupId>
             <artifactId>persistence</artifactId>
         </dependency>
+        <dependency>
```
Comment (DerCed) [inline application/pom.xml:23]: So we are aiming for a keycloak specific integration?

File: webclient/app/public/logout.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
...].exports; var t = n[r] = { i: r, l: !1, exports: {} }; return e[r].call(t.exports, t, t.exports, p), t.l = !0, t.exports } p.m = e, p.c = n, p.d = function (e, r, t) { p.o(e, r) || Object.defineProperty(e, r, { enumerable: !0, get: t }) }, p.r = function (e) { "undefined" != typeof Symbol && Symbol.toStringTag && Object.defineProperty(e, Symbol.toStringTag, { value: "Module" }), Object.defineProperty(e, "__esModule", { value: !0 }) }, p.t = function (e, r) { if (1 & r && (e = p(e)), 8 & r) return e; if (4 & r && "object" == typeof e && e && e.__esModule) return e; var t = Object.create(null); if (p.r(t), Object.defineProperty(t, "default", { enumerable: !0, value: e }), 2 & r && "string" != typeof e) for (var n in e) p.d(t, n, function (r) { return e[r] }.bind(null, n)); return t }, p.n = function (e) { var r = e && e.__esModule ? function () { return e.default } : function () { return e }; return p.d(r, "a", r), r }, p.o = function (e, r) { return Object.prototype.hasOwnProperty.call(e, r) }, p.p = "./"; var l = this.webpackJsonpapp = this.webpackJsonpapp || [], a = l.push.bind(l); l.push = r, l = l.slice(); for (var f = 0; f < l.length; f++)r(l[f]); var i = a; t() }([])</script>
```
Comment (DerCed) [inline webclient/app/public/logout.html:18]: What is this file for?

**Review by DerCed — APPROVED** [review]
LGTM - But we have to look in the frontend integration at a later time

File: webclient/app/public/logout.html (hunk 2 above)
Comment (witchpou) [reply webclient/app/public/logout.html:18]: has to be deleted - thanks.

File: application/pom.xml (hunk 1 above)
Comment (witchpou) [reply application/pom.xml:23]: it was just the best-working way...

**Pull request merged**